lib/compiler.js

```diff
@@ -91,12 +97,41 @@ module.exports = function compile(options) {
     models: modelInstructions,
     middleware: middlewareInstructions,
     components: componentInstructions,
+    mixins: mixinInstructions,
     files: {
       boot: bootScripts
     }
   });
 };
 
+function buildAllMixinInstructions(appRootDir, mixinSources, options) {
+  var extensions = _.without(_.keys(require.extensions), '.json');
+  var files = options.mixins || [];
+  mixinSources.forEach(function(dir) {
+    dir = path.resolve(appRootDir, dir);
```

Comment:
This should use the recently added method `resolveAppPath` here in order to correctly support sources pointing to modules. See also the way how module sources are resolved in https://github.com/strongloop/loopback-boot/blob/d7c67c803aadd46a514114f449fffabe3edf110e/lib/compiler.js#L274-L280.

Here is what I'd like to see: a shared function that can be used by both `findModelDefinitions` and `buildAllMixinInstructions` for converting a list of source dirs into a list of script files to consider.

It's important to keep the mixin behaviour consistent with models. Few cases to keep in mind:
 - `mixinSources: ['fabien-helpers/mixins']` should scan the directory `node_modules/fabien-helpers/mixins`
 - `mixinSources: ['fabien-helper/server/mixins', fabien-helper/common/mixins']` should not throw even when none of the directories exist. (We can discuss this if you disagree.)